### PR TITLE
[Layout] Adds Row and Column Layout components

### LINKED ISSUE #1111: [Layout] Create a RowLayout component
At the moment Cosmos provides a set of components but lacks a system to organize elements on the screen.

The `RowLayout` component allows positioning sections of the page into horizontal rows.

The `RowLayout` component is a grid container that wraps its immediate children into `RowLayout.Item`s and separates them with a grid gap.

Other alternatives are using margins, but since we are not supporting IE11 a grid gap offers a clear implementation.

The gutter between rows allows 4 sizes: none, default, small and large.

The name `Row` has raised some question, it's worth testing it.

All layouts should be responsive (or ready to be responsive), accessible and documented.

**Important:** In order to have consistent spacing across the system, the immediate children of the `RowLayout.Item` should have it's margin cleared. @francocorreasosa suggested to make this by default but optional, in case the author wants to keep them.

### LINKED ISSUE #1112: [Layout] Create a ColumnLayout component
At the moment Cosmos provides a set of components but lacks a system to organize elements on the screen.

The ColumnLayout component allows positioning sections of the page into vertical columns.

The ColumnLayout component is a grid container that wraps its immediate children into ColumnLayout.Items and separates them with a grid gap.

The gutter between rows allows 4 sizes: none, default, small and large.

The `ColumnLayout` allows splitting the container into up to 4 columns.

All layouts should be responsive (or ready to be responsive), accessible and documented.

Important: In order to have consistent spacing across the system, the immediate children of the ColumnLayout.Item should have it's margin cleared. @francocorreasosa suggested to make this by default but optional, in case the author wants to keep them.

### PR DESCRIPTION
**Pull request by andresgalante**
This PR introduces Column and Row Layout components described on #1111  and #1112

@siddharthkp 2 questions for you:

1. should we merge this into a new feature branch where we can also merge the other layouts and the margin PR and play with them?
1. Is it possible to create a new section in the docs for the layouts?

Closes #1111 
Closes #1112

### CODE REVIEW

File: core/components/molecules/column-layout/column-layout.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  --ColumnLayout--Default--GridGap: ${spacing.medium};
+  --ColumnLayout--Spacious--GridGap: ${spacing.large};
+  --ColumnLayout--Header--Color: ${spacing.large};
+
+  display: grid;
+  grid-gap: ${props => gutterOptions[props.gutter]};
+  grid-template-columns: 1fr;
+  
+  /* Placeholder width media feature until we have global variables for breakpoints */
+  @media (min-width: 768px) {
```
Comment (francocorreasosa) [inline core/components/molecules/column-layout/column-layout.js]: Can we make a token for this?

File: examples/manage/public/index.html [hunk 2]
```diff
@@ -6,5 +6,5 @@
   <body>
     <div id="root"></div>
     <script src="main.js"></script>
-  </body>
+  <script type="text/javascript" src="main.d13d9e79a1a5a27ced6b.js"></script></body>
```
Comment (francocorreasosa) [inline examples/manage/public/index.html]: Oops this should not be here 🙈

File: internal/docs/public/index.html [hunk 3]
```diff
@@ -15,5 +15,5 @@
   </head>
   <body>
     <div id="root"></div>
-  </body>
+  <script type="text/javascript" src="main.b291feb9db13afb3a38d.js"></script></body>
```
Comment (francocorreasosa) [inline internal/docs/public/index.html]: This should not be here either

File: core/components/molecules/column-layout/column-layout.js (hunk 1 above)
Comment (andresgalante) [reply core/components/molecules/column-layout/column-layout.js]: Yes we should and will when we tackle the variable system later this year

**Pull request merged**